rust/db_handling/src/identities.rs

```diff
@@ -280,13 +280,39 @@ pub fn create_increment_set(increment: u32, multisigner: &MultiSigner, network_s
 }
 
 /// Check derivation format and determine whether there is a password
-pub fn check_derivation_format(path: &str) -> Result<bool, ErrorSigner> {
+pub fn is_passworded(path: &str) -> Result<bool, ErrorSigner> {
     match REG_PATH.captures(path) {
         Some(caps) => Ok(caps.name("password").is_some()),
         None => return Err(ErrorSigner::AddressGeneration(AddressGeneration::Extra(ExtraAddressGenerationSigner::InvalidDerivation))),
     }
 }
 
+/// Enum to describe derivation proposed
+pub enum DerivationCheck {
+    BadFormat, // bad format, still gets json back into UI
+    Password, // passworded, no dynamic checking for collisions
+    NoPassword(Option<(MultiSigner, AddressDetails)>), // no password and optional collision
+}
+
+/// Function to check if proposed address collides,
+/// i.e. if the proposed path for given seed name and network specs key already exists
+pub fn derivation_check (seed_name: &str, path: &str, network_specs_key: &NetworkSpecsKey, database_name: &str) -> Result<DerivationCheck, ErrorSigner> {
+    match is_passworded(path) {
+        Ok(true) => Ok(DerivationCheck::Password),
+        Ok(false) => {
+            let mut found_collision = None;
+            for (multisigner, address_details) in get_all_addresses(database_name)?.into_iter() {
+                if (address_details.seed_name == seed_name)&&(address_details.path == path)&&(address_details.network_id.contains(network_specs_key))&&(!address_details.has_pwd) {
+                    found_collision = Some((multisigner, address_details));
+                    break;
+                }
+            }
+            Ok(DerivationCheck::NoPassword(found_collision))
+        },
+        Err(_) => Ok(DerivationCheck::BadFormat)
```

Comment:
I am not sure about propagating an `Err` situation as an `Ok` situation.

Reply:
This is actually supposed to be idiomatic - the function just throws when derivation format is invalid, which is a valid result here; errors are more of disk access nature and db corruption, that are unreachable in this branch of match. This function is used more than in one place, for example, in interface_signer line 245 below, where it is not propagated as error but just defines UI representation.